sotoki/sotoki.py

```diff
@@ -840,27 +840,30 @@ def download(url, output, timeout=None):
 
 def get_filetype(headers, path):
     ftype = "none"
+    known_type_found = 0
     if "content-type" in headers:
         if ("png" in headers["content-type"]) or ("PNG" in headers["content-type"]):
             ftype = "png"
+            known_type_found = 1
         elif (
             ("jpg" in headers["content-type"])
             or ("jpeg" in headers["content-type"])
             or ("JPG" in headers["content-type"])
             or ("JPEG" in headers["content-type"])
         ):
             ftype = "jpeg"
+            known_type_found = 1
         elif ("gif" in headers["content-type"]) or ("GIF" in headers["content-type"]):
             ftype = "gif"
-    else:
-        with magic.Magic() as m:
-            mine = m.id_filename(path)
-            if "PNG" in mine:
-                ftype = "png"
-            elif "JPEG" in mine:
-                ftype = "jpeg"
-            elif "GIF" in mine:
-                ftype = "gif"
+            known_type_found = 1
+    elif (known_type_found == 0):
```

Comment:
It's should be a "if".
That way if content-type is not found in headers, it fall back to magic

Reply:
Sorry. My bad. Such a small mistake.